presto-orc/src/main/java/com/facebook/presto/orc/metadata/DwrfStripeCacheMode.java

```diff
@@ -19,14 +19,65 @@
 public enum DwrfStripeCacheMode
 {
     // no cache
-    NONE,
+    NONE(false, false),
 
     // cache contains stripe indexes
-    INDEX,
+    INDEX(true, false),
 
     // cache contains stripe footers
-    FOOTER,
+    FOOTER(false, true),
 
     // cache contains stripe indexes and footers
-    INDEX_AND_FOOTER;
+    INDEX_AND_FOOTER(true, true);
+
+    private final boolean hasIndex;
+    private final boolean hasFooter;
+
+    DwrfStripeCacheMode(boolean hasIndex, boolean hasFooter)
```

Comment:
should this be private constructor ?

Reply:
If no access modifier is specified enum defaults to private constructor, so no changes is required here. https://stackoverflow.com/a/7748220

Reply:
You are right. Enums are sealed, so making it private is redundant.